clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTaskTest.java

```diff
@@ -868,20 +868,32 @@ private void prepareAggKafkaConsumerServiceMock() {
     }).when(aggKafkaConsumerService).hasConsumerAssignedFor(anyString(), anyString(), anyString(), anyInt());
 
     doAnswer(invocation -> {
+      String versionTopic = invocation.getArgument(0, String.class);
       Set<TopicPartition> topicPartitions = invocation.getArgument(1, Set.class);
-      inMemoryLocalKafkaConsumer.batchUnsubscribe(topicPartitions);
-      inMemoryRemoteKafkaConsumer.batchUnsubscribe(topicPartitions);
+      /**
+       * The internal {@link SharedKafkaConsumer} has special logic for unsubscription to avoid some race condition
+       * between the fast unsubscribe and re-subscribe.
+       * Please check {@link SharedKafkaConsumer#unSubscribe} to find more details.
+       */
+      localKafkaConsumerService.batchUnsubscribe(versionTopic, topicPartitions);
+      remoteKafkaConsumerService.batchUnsubscribe(versionTopic, topicPartitions);
```

Comment:
I'm a bit confused here... the newly added comment talks about `SKC:: unSubscribe` which is what the code used to call, but then this PR replaces the call to this for `KCS::batchUnsubscribe`... so should the comment mention the new function which is called instead of the old one? Or is this comment intended to be a warning about how the code was before (in which case probably some rephrasing is needed)...?

Reply:
Actually, neither `inMemoryLocalKafkaConsumer` nor `inMemoryRemoteKafkaConsumer` is `SharedKafkaConsumer` since they are in-memory impl of `KafkaConsumerWrapper`, and that is the root cause of the flaky tests.
`SharedKafkaConsumer#unSubscribe`/ `SharedKafkaConsumer#batchUnsubscribe`has logic to prevent the race condition, and change the invocation to `KafkaConsumerService`, which will internally tigger the invocations of the corresponding `SharedKafkaConsumer` functions.

Reply:
Ok so I think the take away from what you're saying is that essentially this was just a test issue, and not a main code issue, thanks to the synchronization in `SKC::unSubscribe` (which the main code calls...). If that is the right understanding, then I think the JavaDoc added here should be clarified a bit... essentially, something along the lines of "`MockInMemoryConsumer` does not have proper synchronization, whereas as `SKC` does, and so instead of calling the mock consumer we're calling the KCS to rely on that one doing the synchronization"...

Reply:
Yes, it is a test-only issue, and the reason I didn't mention `MockInMemoryConsumer` explicitly since it will be removed from the new code.
I can add some javadoc related to `MockInMemoryConsumer` as you mentioned to let other developers know why `MockInMemoryConsumer` won't work here.